proposals/yield-in-try-catch.md

```diff
@@ -334,52 +362,135 @@ bool MoveNext()
 
 ## Considerations
 
-### yield inside catch with nested finally
-
-[catch-finally]: #yield-inside-catch-with-nested-finally
-
-The `yield` statement cannot be reasonbly supported inside a `catch` blocks with a nested `finally` due to the behavior of the `Dispose` method. It is possible that a `catch` block will run as part of executing a `finally` in the `Dispose` method.
+### yield inside finally
 
-For example consider the following:
+Consideration was given to supporting `yield` inside a `finally` block but this creates significant semantic challenges. Consider the following code as an example:
 
 ```csharp
-var e = InCatchFinally();
+var e = M().GetEnumerator();
 e.MoveNext();
 e.Dispose();
 
-IEnumerable<int> InCatchFinally()
+void M()
+{
+    try
+    {
+        yield return 42;
+    }
+    finally
+    {
+        yield return 13;
+    }
+}
+```
+
+The `finally` block would be executed in `Dispose` which means the language has to decide on the behavior of `yield` during `Dispose`. That could be modeled as:
+
+1. Throw an exception when `yield` is encountered in `Dispose`.
+2. Silently ignore the value and continue executing the `finally` block without suspend.
+
+Neither of these seem like desirable outcomes and as such `yield` will not be allowed in `finally` blocks.
+
+## Open Issues
+
+### Preserve catch in Dispose paths
+
+The `Dispose` method could include mirroring both `catch` and `finally` blocks. This would allow the `catch` block to be executed in `Dispose` when `finally` blocks threw an exceptoin. The approach for this would be to do the followng.
```

Comment:
```suggestion
The `Dispose` method could include mirroring both `catch` and `finally` blocks. This would allow the `catch` block to be executed in `Dispose` when `finally` blocks threw an exception. The approach for this would be to do the following.
```